src/satellite/config.ts

```diff
@@ -34,74 +34,86 @@ export const satelliteDefaults: SatelliteOpts = {
   minSnapshotWindow: 40
 }
 
-export interface SatelliteClientOpts {
-  app: string
-  env: string
-  token: string
-  port: number
-  address: string
-  timeout: number
-  pushPeriod: number
-}
-
 export const satelliteClientDefaults = {
+  env: "default",
   timeout: 3000,
-  pushPeriod: 500
+  pushPeriod: 500,
+  insecure: false
 }
 
-export interface SatelliteClientOverrides {
+const baseDomain = "electric-sql.com"
+
+export interface SatelliteClientOpts {
   app: string
-  env: string
+  env?: string
   token: string
+  host: string
   port: number
-  address: string
   timeout?: number
   pushPeriod?: number
+  insecure?: boolean
 }
 
+
 // Config spec
 export interface ElectricConfig {
   app: string
-  env: string
+  env?: string
   token: string
   migrations?: Migration[],
   replication?: {
-    address: string
+    host: string
     port: number
+    insecure: boolean
   }
   debug?: boolean,
 }
 
+const electricConfigDefaults: Partial<ElectricConfig> = {
+  env: "default"
+}
+
+export const addDefaultsToElectricConfig = (config: ElectricConfig): ElectricConfig => {
+  const host = (config.replication?.host) ?? `${config.env}.${config.app}.${baseDomain}`
+  const port = (config.replication?.port) ?? 443
+  const insecure = (config.replication?.insecure) ?? false
 
-export const validateConfig = (config : any) => {
+  config = {
+    ...electricConfigDefaults,
+    ...config
+  }
+  config.replication = { ...config.replication, host, port, insecure }
+
+  return config
```

Comment:
Let's not do argument reassignment, that way lies JS madness :D

```suggestion
  const fullConfig: ElectricConfig = {
    ...electricConfigDefaults,
    ...config,
    replication: { ...config.replication, host, port, insecure }
  }

  return fullConfig
```

Reply:
but looks so harmless and is so tempting :). Will fix that.

Reply:
done